ArchiSteamFarm/IPC.cs

```diff
@@ -509,6 +513,91 @@ internal static class IPC {
+				return true;
+			}
+
+			IEnumerable<Task<bool>> tasks = bots.Select(bot => Task.Run(() => bot.DeleteRedeemedKeysFiles()));
+			ICollection<bool> results;
+
+			switch (Program.GlobalConfig.OptimizationMode) {
+				case GlobalConfig.EOptimizationMode.MinMemoryUsage:
+					results = new List<bool>(bots.Count);
+					foreach (Task<bool> task in tasks) {
+						results.Add(await task.ConfigureAwait(false));
+					}
+
+					break;
+				default:
+					results = await Task.WhenAll(tasks).ConfigureAwait(false);
+					break;
+			}
+
+			if (results.Any(result => !result)) {
+				await ResponseJsonObject(request, response, new GenericResponse<object>(false, "Removing one or more files failed, check ASF log for details"), HttpStatusCode.BadRequest).ConfigureAwait(false);
+				return true;
+			}
+
+			await ResponseJsonObject(request, response, new GenericResponse<object>(true, "OK")).ConfigureAwait(false);
+			return true;
+		}
+
+		private static async Task<bool> HandleApiGamesToRedeemInBackgroundGet(HttpListenerRequest request, HttpListenerResponse response, string[] arguments, byte argumentsIndex) {
+			if ((request == null) || (response == null) || (arguments == null) || (argumentsIndex == 0)) {
+				ASF.ArchiLogger.LogNullError(nameof(request) + " || " + nameof(response) + " || " + nameof(arguments) + " || " + nameof(argumentsIndex));
+				return false;
+			}
+
+			if (arguments.Length <= argumentsIndex) {
+				return false;
+			}
+
+			string argument = WebUtility.UrlDecode(string.Join("", arguments.Skip(argumentsIndex)));
+
+			HashSet<Bot> bots = Bot.GetBots(argument);
+			if ((bots == null) || (bots.Count == 0)) {
+				await ResponseJsonObject(request, response, new GenericResponse<object>(false, string.Format(Strings.BotNotFound, argument)), HttpStatusCode.BadRequest).ConfigureAwait(false);
+				return true;
+			}
+
+			IEnumerable<Task<(Dictionary<string, string> Used, Dictionary<string, string> Unused)>> tasks = bots.Select(bot => bot.GetUsedAndUnusedKeys());
+			ICollection<(Dictionary<string, string> Used, Dictionary<string, string> Unused)> results;
+
+			switch (Program.GlobalConfig.OptimizationMode) {
+				case GlobalConfig.EOptimizationMode.MinMemoryUsage:
+					results = new List<(Dictionary<string, string> used, Dictionary<string, string> unused)>();
+					foreach (Task<(Dictionary<string, string> used, Dictionary<string, string> unused)> task in tasks) {
+						(Dictionary<string, string> Used, Dictionary<string, string> Unused) result = await task.ConfigureAwait(false);
+
+						results.Append(result);
+					}
+
+					break;
+				default:
+					results = (await Task.WhenAll(tasks).ConfigureAwait(false));
+					break;
+			}
+
+			IEnumerable<GamesToRedeemInBackgroundResponse> responses = results.Select(result => new GamesToRedeemInBackgroundResponse(result.Used, result.Unused));
+
+			await ResponseJsonObject(request, response, new GenericResponse<object>(true, "OK", responses)).ConfigureAwait(false);
```

Comment:
This should not be `object` as it can't be strong-typed. Instead you should probably use `Dictionary<string, (Dictionary <string, string> UsedKeys, Dictionary<string, string> UnusedKeys)>` where first `key` will be name of the bot. Right now your array is entirely useless as it doesn't specify which response is for which bot.

Reply:
Where do we get the bot-name from? the property in Bot is private. change it?

Reply:
sure. your comment wasn't displayed for me 
>Although you need to be smart about this as ToDictionary is not lazy-evaluated, so it might make more sense to instead do something like bots.Select(bot => (bot.botName, bot.GetUsedAndUnusedKeys()).

Reply:
Then you can just make an array of them and get it done, while in the next step you can generate proper response out of both parts.